docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,151 @@
+  * Add also, add the `baremetalhost.metal3.io/paused` annotation:
+    * This is required since once the annotation is added, BMO updates the BMH's status and delete the annotation.
+    * The [paused](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#pausing-reconciliation) annotation, is similar to the `detached` annotation we're already using - but, it's also pausing the reconcile loop. Thus, keeping the status annotation intact.
+
+For BMHs in the passive hub: (In assisted-service -> `bmh_agent_controller`)
+* When the BMH is applied, the status will be missing - since reconcile is paused.
+  * To restore the original status:
+    * Remove the `baremetalhost.metal3.io/paused` annotation
+      * Status is restored, but `baremetalhost.metal3.io/status` annotation is removed.
+    * Hence, add the status and paused annotation back
+      * By detecting a BMH with `provisioned` state and a missing status annotation.  
+
+#### Solution for deleted Agent CRs
+
+1. Don't delete the Agent CR when an associated host doesn't exits on DB.
+   * Remove relevant [code](https://github.com/openshift/assisted-service/blob/5bbf696fc0cc5a220fd8a0d1e408ab0bcc889e3f/internal/controller/controllers/agent_controller.go#L143-L151) from AgentController.
+
+2. Restore the Host record in DB according to properties from the Agent CR.
+   * Seems we can infer more than the mandatory properties:
+     * host.ID <-- agent.Name
+     * host.Role <-- agent.Role
+     * host.InfraEnvID <-- agent.metadata.annotations['infraenvs.agent-install.openshift.io']
+     * host.Kind <-- 'Host' or 'AddToExistingClusterHost' if cluster is already installed
+     * host.InstallerArgs <-- agent.InstallerArgs
+     * host.RequestedHostname <-- agent.Hostname
+   * Some of the inventory data can be restored from annotations.
+     E.g.:
+     * inventory.agent-install.openshift.io/cpu-architecture: x86_64
+     * inventory.agent-install.openshift.io/cpu-virtenabled: "true"
+     * inventory.agent-install.openshift.io/host-isvirtual: "true"
+     * inventory.agent-install.openshift.io/host-manufacturer: RedHat
+     * inventory.agent-install.openshift.io/host-productname: KVM
+     * inventory.agent-install.openshift.io/storage-hasnonrotationaldisk: "false"
+   * Note: We might need to add more annotations for additional inventory data.
+   
+### Risks and Mitigations
+
+* In general, these changes might affect flows on spoke cluster (before and after restore), need to ensure no regressions on all scenarios.
+* Adding the 'paused' annotation means a stale BMH status, need to check if it's crucial.
+* Keeping the Agent CR and restoring values into Host in DB could affect current flows (e.g. BindHost, UnbindHost, etc).
+* Restoring Hosts will result in partial records in DB. E.g. missing Inventory - could be challenging to tackle (I.e. probably need to add annotations to Agent).
+
+### Open Questions
+
+* Any specific concerns for the ConvergedFlow/Non-ConvergedFlow?
```

Comment:
If it was the converged flow, would we be concerned about how the node is registered in Ironic? Like if the state of the node in Ironic somehow wasn't preserved after restoring and we can't deprovision (or scale down) the host

Reply:
I think everything they need should be in the BMH. So as long as that is restored correctly it should be fine. :pray:

Reply:
I hope so 😅 I just recalled times where the BMH would either be stuck "deprovisioning" or "inspecting" when ironic was reporting "clean wait" or something else